client/sniffer/src/test/java/org/opensearch/client/sniff/OpenSearchNodesSnifferParseTests.java

```diff
@@ -163,33 +135,23 @@ public void testParsingPublishAddressWithES7Format() throws IOException {
         assertEquals("http", nodes.get(0).getHost().getSchemeName());
     }
 
-    private Node node(int port, String name, String version, boolean master, boolean data, boolean ingest) {
-        final Set<String> roles = new TreeSet<>();
-        if (master) {
-            roles.add("master");
-        }
-        if (data) {
-            roles.add("data");
-        }
-        if (ingest) {
-            roles.add("ingest");
-        }
-        return node(port, name, version, roles);
-    }
-
     private Node node(int port, String name, String version, String... roles) {
         return node(port, name, version, new TreeSet<>(Arrays.asList(roles)));
     }
 
     private Node node(int port, String name, String version, Set<String> roles) {
         HttpHost host = new HttpHost("127.0.0.1", port);
         Set<HttpHost> boundHosts = new HashSet<>(2);
-        boundHosts.add(host);
         boundHosts.add(new HttpHost("[::1]", port));
+        boundHosts.add(host);
         Map<String, List<String>> attributes = new HashMap<>();
         attributes.put("dummy", singletonList("everyone_has_me"));
         attributes.put("number", singletonList(name.substring(1)));
-        attributes.put("array", Arrays.asList(name.substring(0, 1), name.substring(1)));
+        attributes.put("array", singletonList(Arrays.asList(name.substring(0, 1), name.substring(1)).toString()));
```

Comment:
It does look like we set now `array` as array of arrays (`singletonList(Arrays.asList(...)`), is it on purpose?

Reply:
Hi @reta , thanks for the review! Let me explain it shortly, and I will seek for a better solution.

Reply:
I probably should update the code of parsing the API response (such as `[m, 1]`) to be a Java array, instead of modifying the unit test. I will have a try.

It comes from a change in Elasticsearch 6.1, commit: https://github.com/elastic/elasticsearch/commit/00dfdf50cfee9142daf987d38adf3eff3a971573, doc: https://www.elastic.co/guide/en/elasticsearch/reference/6.1/release-notes-6.1.0.html#breaking-6.1.0

When adding the setting into yml config file:
`node.attr.array: 0, [m, 1]`

There is a change of the presentation of array value in the response of Nodes Info API:
In Elasticsearch 6.1 and above:
```
      "attributes": {
        "array": "[m, 1]"
      },
```

In Elasticsearch 6.0 and below:
```
      "attributes": {
        "array.0": "m",
        "array.1": "1"
      },
```

The current Sniffer code, can only parse the old format.

Reply:
Oh I seem thank you for explaining @tlfeng

Reply:
I updated the API response parser to meet the array presentation in ES 6.1 for the node attributes setting.